src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -1095,16 +1095,14 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
                 throw CreateRetryException();
             }
 
-            Http2Stream http2Stream = null;
-
             try
             {
-                http2Stream = AddStream(request);
-                int streamId = http2Stream.StreamId;
-
                 // Generate the entire header block, without framing, into the connection header buffer.
                 WriteHeaders(request);
 
+                Http2Stream http2Stream = AddStream(request);
+                int streamId = http2Stream.StreamId;
```

Comment:
If something fails between here and the return, such as StartWriteAsync getting canceled, what removes the newly added stream?

Reply:
Yeah, there's a problem here. I changed this as I was changing all the RemoveStream stuff. But I think it ended up wrong.

I'm a little worried there are some error paths in this method that aren't handled properly. I'll go through it and try to ensure all cases are handled.